### PR TITLE
Add field to ThetaSketch post aggregation. Allow dimensions to be null

### PR DESCRIPTION
**Pull request by st0ckface**
Add errorBoundsStdDev to ThetaSketch postAggregation. Allow dimensions to be null in GroupByQuery as they are not required.

With dimensions needing to be NonNull, I've been writing a bunch of code like:
` DruidGroupByQuery.builder().dimensions(Collections.emptyList())....`

### CODE REVIEW

**Comment by abhi-zapr** [comment]
Hey @st0ckface ,
Thank you for your contribution.
We will review the PR by the end of the week.
Thanks.

**Comment by patilvikram** [comment]
@st0ckface As per druid documentation, dimensions should be provided.  Its mandatory field in the query. But if you found out during testing it works without dimensions you should report to druid . I believe these are dimensions used for grouping. http://druid.io/docs/latest/querying/groupbyquery.html

**Comment by st0ckface** [comment]
Sure, I see that in the documentation. It works in practice, I've posted to the google group to clarify. Maybe there's a reason you wouldnt want to even if its a valid query

**Comment by abhi-zapr** [comment]
I see that @patilvikram already cleared that dimensions should be provided, and hence mandatory field. Thanks @patilvikram .
So you should update the PR to exclude that part.

And regarding including **errorBoundsStdDev** field in ThetaSketchEstimatePostAggregator, thanks for contributing this. This field was not in the druid documentation and hence we didn't include. But it is there in source code, so we should include it. Update the PR to address the review comments.

Thanks.

**Comment by st0ckface** [comment]
sure, i can make those changes. Interesting though, the druid documentation that was linked to here:
http://druid.io/docs/latest/querying/groupbyquery.html
Has a disclaimer at the top of the page which talks about the scenario where  Time is the only grouping, in which case a timeSeriesQuery would be more appropriate. That implies that dimensions can be null, but in that case a groupByQuery is a less suitable option.

Either way, I'll incorporate the feed back and push something up soon.

File: src/test/java/in/zapr/druid/druidry/extensions/datasketches/postAggregator/ThetaSketchEstimatePostAggregatorTest.java [hunk 1]
```diff
@@ -73,6 +73,30 @@ public void testAllFieldsWithFieldAccess() throws JsonProcessingException, JSONE
 
     }
 
+    @Test
+    public void testAllFieldsWithFieldAccess() throws JsonProcessingException, JSONException {
+
+        FieldAccessPostAggregator fieldAccessPostAggregator =
+                new FieldAccessPostAggregator("stars");
+
+        ThetaSketchEstimatePostAggregator thetaSketchEstimatePostAggregator =
```
Comment (abhi-zapr) [inline src/test/java/in/zapr/druid/druidry/extensions/datasketches/postAggregator/ThetaSketchEstimatePostAggregatorTest.java]: You need to update all the test cases with this new field and with constructor changed to builder.

**Comment by abhi-zapr** [comment]
Hey,
@st0ckface can you address the pending review [comment](https://github.com/zapr-oss/druidry/pull/66#discussion_r240002125), then we can merge your PR.

**Comment by abhi-zapr** [comment]
Hey @st0ckface merging this PR in feature-branch and I will be applying test case changes required as per review [comment](https://github.com/zapr-oss/druidry/pull/66#discussion_r240002125) so that we can include this patch in next release.

File: src/test/java/in/zapr/druid/druidry/extensions/datasketches/postAggregator/ThetaSketchEstimatePostAggregatorTest.java (hunk 1 above)
Comment (abhi-zapr) [reply src/test/java/in/zapr/druid/druidry/extensions/datasketches/postAggregator/ThetaSketchEstimatePostAggregatorTest.java]: Approving and merging into separate feature branch to fix test cases.

**Pull request merged**